### DISCUSSION TITLE
Is it possible to do a multiarch build with linux/arm/v7?

### DISCUSSION THREAD

**Discussion by BradAF** [abiosoft/colima#667] [Q&A] [answered]
I'm on an Intel Mac using colima, and I have successfully been able to build an image for linux/arm64 alongside my linux/386 builds using:

```sh
docker buildx create \
  --name multiarch \
  --platform linux/amd64,linux/amd64/v2,linux/amd64/v3,linux/arm64,linux/386,linux/arm/v7 \
  --use colima

docker build \
  --push \
  --platform linux/arm64/v8,linux/amd64 \
  --tag ghcr.io/myname/myimage:v0.2 \
  .
```

However, when trying to create an image for an old raspberry pi, doing:

```sh
docker build \
  --push \
  --platform linux/arm/v7,linux/arm64/v8,linux/amd64 \
  --tag ghcr.io/myname/myimage:v0.2 \
  .
```

fails with:
>  \> [linux/arm/v7 frontend-build-stage 4/6] RUN npm ci:
> #0 0.122 exec /bin/sh: exec format error

Is this because colima does not support linux/arm/v7, or am I missing something?

**Comment by abiosoft** [comment, marked as answer]
Yeah, `linux/arm/v7` it is currently not bundled.

You can actually just run the following and it should work afterwards.

```
docker run --privileged --rm tonistiigi/binfmt --install all
```

**Reply by BradAF** [reply]
Great, thanks!

I ran that and was able to at least get the RUN command to start. The problem now is that after 7 hours or so, it still hasn't finished a simple `npm ci`. As far as I can tell, everything else is correct regarding the docker-container builder but now when I try to stop colima it just gives an error even when just starting it up and immediately stopping it which it didn't used to do before:

```sh
❯ docker buildx ls
NAME/NODE     DRIVER/ENDPOINT  STATUS   BUILDKIT PLATFORMS
multiarch *   docker-container
  multiarch0  colima           inactive          linux/amd64*, linux/amd64/v2*, linux/amd64/v3*, linux/arm64*, linux/386*, linux/arm/v7*
colima        docker
  colima      colima           running  20.10.20 linux/amd64, linux/arm64, linux/386, linux/riscv64, linux/ppc64le, linux/s390x, linux/arm/v7, linux/arm/v6
default                        error
desktop-linux                  error
```

```sh
❯ colima stop
INFO[0000] stopping colima
INFO[0000] stopping ...                                  context=docker
INFO[0001] stopping ...                                  context=vm
> [hostagent] goroutine 17 [running, locked to thread]:
> [hostagent] runtime/cgo.Handle.Value(...)
> [hostagent] 	/usr/local/Cellar/go/1.20.1/libexec/src/runtime/cgo/handle.go:124
> [hostagent] github.com/Code-Hex/vz/v3.changeStateOnObserver(0x0, 0xc000006601?)
> [hostagent] 	/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/!code-!hex/vz/v3@v3.0.4/virtualization.go:160 +0xec
```
> \> did not receive an event with the "exiting" status
> FATA[0181] error stopping vm: error at 'stopping': exit status 1

I'll keep troubleshooting, but thanks for pointing me in the right direction!

**Reply by abiosoft** [reply]
The error is a known issue upstream in Lima https://github.com/lima-vm/lima/issues/1381.
It is pretty harmless less and you can simply discard with <kbd>Ctrl+C</kbd>.
